src/linked_list.py

```diff
@@ -16,6 +22,9 @@ def push(self, contents):
 
     def pop(self):
         """Remove and return the current head node."""
+        if not self.head_node:
+            print("Linked list is already empty")
```

Comment:
Again, rather than just print a message, this is where you should raise an exception too let the user know that something has gone wrong. Remember, this could be in a scenario where thousands of Linked Lists are being used in the depths of a very complex program. The designer of that program probably won't see the printed message, so an exception is the correct way to communicate. And this exception can be handled in a way that makes sense to the user's program needs.